src/service.rs

```diff
@@ -116,28 +130,71 @@ impl Service {
     fn export_listener(&self) -> Result<Listener> {
         let mut filters = Vec::new();
 
-        // @TODO no way, move this to a function or something.
-        let mut buf = Vec::new();
-        prost::Message::encode(
-            &Router {
-                ..Default::default()
-            },
-            &mut buf,
-        )?;
+        fn encode(arg: impl prost::Message) -> Vec<u8> {
+            let mut buf = Vec::new();
+            prost::Message::encode(&arg, &mut buf).unwrap();
+            buf
+        }
 
         let config = prost_types::Any {
             type_url: "type.googleapis.com/envoy.extensions.filters.http.router.v3.Router"
                 .to_string(),
-            value: buf,
+            value: encode(Router {
+                ..Default::default()
+            }),
+        };
+
+        // WASM section, @TODO move out to a new method
+        let wasm_filter = Wasm {
+            config: Some(PluginConfig {
+                name: format!("Service::{:?}", self.id),
+                root_id: format!("Service::{:?}", self.id),
+                vm: Some(Vm::VmConfig(VmConfig {
+                    vm_id: format!("Service::{:?}", self.id),
+                    runtime: "envoy.wasm.runtime.v8".to_string(),
+                    configuration: Some(prost_types::Any {
+                        type_url: "type.googleapis.com/google.protobuf.StringValue".to_string(),
+                        value: encode(serde_json::to_string(&self.clone()).unwrap()),
```

Comment:
Does this type require to call `encode` here? See below for another comment referring to this.

Also, use `?` rather than `unwrap()`. This could end up being:
```rust
value: encode(serde_json::to_string(...)?)?,
```